pallets/simple-staking/src/lib.rs

```diff
@@ -0,0 +1,302 @@
+			if allowed > T::MaxAuthors::get() {
+				log::warn!(
+					"allowed > T::MaxAuthors; you might need to run benchmarks again"
+				);
+			}
+			<AllowedAuthors<T>>::put(&allowed);
+			Self::deposit_event(Event::NewAllowedAuthorCount(allowed));
+			Ok(().into())
+		}
+
+		#[pallet::weight(T::WeightInfo::set_author_bond())]
+		pub fn set_author_bond(origin: OriginFor<T>, author_bond: BalanceOf<T>) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			<AuthorBond<T>>::put(&author_bond);
+			Self::deposit_event(Event::NewAuthorBond(author_bond));
+			Ok(().into())
+		}
+
+		// 1. worse case
+		// #[pallet::weight(T::WeightInfo::register_as_author(T::MaxAuthors::get()))]
+		// 2. accurate, provided by user, checked by chain
+		// #[pallet::weight(T::WeightInfo::register_as_author(authors_count))]
+		// pub fn register_as_author(origin: OriginFor<T>, authors_count: u32) -> DispatchResultWithPostInfo {}
+		// 3. refund
+		// #[pallet::weight(T::WeightInfo::register_as_author(T::MaxAuthors::get()))]
+		// in case of successful transaction, return `Ok(Some(refund).into())`
+
+		#[pallet::weight(T::WeightInfo::register_as_author(T::MaxAuthors::get()))]
+		pub fn register_as_author(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			// lock deposit to start or require min?
+			let who = ensure_signed(origin)?;
+			let deposit = <AuthorBond<T>>::get();
+			let length = <Authors<T>>::decode_len().unwrap_or_default();
+			ensure!((length as u32) < Self::allowed_authors(), Error::<T>::TooManyAuthors);
+			let new_author = AuthorInfo {
+				who: who.clone(),
+				deposit,
+				last_block: None
+			};
+			let author_count = <Authors<T>>::try_mutate(|authors| -> Result<usize, DispatchError> {
+									if authors.into_iter().any(|author| author.who == who) {
+										Err(Error::<T>::AlreadyAuthor)?
+									} else {
+										T::Currency::reserve(&who, deposit)?;
+										authors.push(new_author);
+										Self::deposit_event(Event::AuthorAdded(who, deposit));
+										Ok(authors.len())
+									}
+								})?;
+			Ok(Some(T::WeightInfo::register_as_author(author_count as u32)).into())
+		}
+
+		#[pallet::weight(T::WeightInfo::leave_intent(T::MaxAuthors::get()))]
+		pub fn leave_intent(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let author_count = <Authors<T>>::try_mutate(|authors| -> Result<usize, DispatchError> {
+									let index = authors.iter().position(|author| author.who == who).ok_or(Error::<T>::NotAuthor)?;
+									T::Currency::unreserve(&who, authors[index].deposit);
+									authors.remove(index);
+									Ok(authors.len())
+								})?;
+			Self::deposit_event(Event::AuthorRemoved(who));
+			Ok(Some(T::WeightInfo::leave_intent(author_count as u32)).into())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub fn account_id() -> T::AccountId {
+			T::TreasuryId::get().into_account()
```

Comment:
```suggestion
			T::TreasuryId::get().into_account()
```

Treasury cna be mistaken with pallet-treasury, I would just call this `pot` and `PotId`.